libparsec/crates/types/src/serialization.rs

```diff
@@ -79,18 +51,6 @@ where
             })
         }
 
-        // Allow decoding of vFF format even if `use-zstd-in-serialization-format` is enabled,
-        // this is needed to be able to communicate between a wasm client and the non-wasm
-        // testbed server (note the trick here is that the data are only produced by the
-        // client, so the client doesn't have to support format v0 as long as there is
-        // only wasm client which is reasonable in a dev scenario).
-        Some(&FORMAT_VFF_VERSION_BYTE) => {
-            rmp_serde::from_slice(&raw[1..]).map_err(|_| DataError::BadSerialization {
-                format: Some(FORMAT_VFF_VERSION_BYTE),
-                step: "msgpack+validation",
-            })
-        }
-
         Some(_) | None => Err(DataError::BadSerialization {
             format: None,
```

Comment:
I would expect that `format` to be set to the invalid value

Reply:
The step that has failed is marked to be `format detection`, then the `format` has to be set to `None` to indicate we are not in any format (i.e. this field is not to indicate what is the format value that is not recognized, I consider there is no need for such information, just like if zstd or msgpack fail de-serialization the specific error message is not provided)